### DISCUSSION TITLE
[markdownlint-rule-helpers] `tokens` property removed from RuleParams in markdownlint 0.34.0

### DISCUSSION THREAD

**Discussion by rsese** [DefinitelyTyped/DefinitelyTyped#69627] [Issues with a @types package]
👋 !  Was having some trouble using the types for markdownlint and markdownlint-rule-helpers where a custom rule we have that uses the `filterTokens()` function:

https://github.com/DavidAnson/markdownlint/blob/b2305efafb034b1f328845aec9928b5363ffd646/helpers/helpers.js#L347

Had a type error saying `Property 'tokens' is missing` in the first argument `params` which is a RuleParams object:

https://github.com/DavidAnson/markdownlint/blob/b2305efafb034b1f328845aec9928b5363ffd646/lib/markdownlint.d.ts#L114-L135

From my poking I found as part of [v0.34.0](https://github.com/DavidAnson/markdownlint/releases/tag/v0.34.0), in https://github.com/DavidAnson/markdownlint/commit/c44f165fe16d299ccfd911e7477318c2ae9846db#diff-bb0111f3d75d5332283370198902c51b2df5924e43692ea7b2e2b64b9cb34acd the `tokens` property was removed from `RuleParams`.

But the version of markdownlint used for the types here specifies `^0.30.0`:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/markdownlint-rule-helpers/package.json#L11

Was going to open a PR to bump the version:

```diff
diff --git a/types/markdownlint-rule-helpers/package.json b/types/markdownlint-rule-helpers/package.json
index 1cc9f3d705..bbbc06ed69 100644
--- a/types/markdownlint-rule-helpers/package.json
+++ b/types/markdownlint-rule-helpers/package.json
@@ -8,7 +8,7 @@
     "dependencies": {
         "@types/markdown-it": "*",
         "@types/node": "*",
-        "markdownlint": "^0.30.0"
+        "markdownlint": "^0.34.0"
     },
     "devDependencies": {
         "@types/markdownlint-rule-helpers": "workspace:."
```

But still new-ish to TS and the types in this repo so figured I would ask here first -- thanks 🙇 !

[a comment by an automated account is not shown]

**Reply by JoshuaKGoldberg** [reply]
I wonder if @DavidAnson should be an owner? I really almost never work in the `markdownlint` area 😬

**Reply by DavidAnson** [reply]
As @rsese notes on a different thread here, I made a specific decision not to document the "-rules" package. I see why having this is useful, but it seems challenging for an unstable API.

**Comment by JoshuaKGoldberg** [comment]
What if the developer using `@types/markdownlint-rule-helpers` is still on an older version of `markdownlint`, say `v0.30.0`. Then requiring `markdownlint` to be `^0.34.0` would break them, no? Has `markdownlint-rule-helpers` stopped supporting older versions of `markdownlint`?

**Reply by rsese** [reply]
ahh yep I wondered about folks on an older version of `markdownlint` and if that meant this would be a breaking change for `@types/markdownlint-rule-helpers`?

> Has markdownlint-rule-helpers stopped supporting older versions of markdownlint?

I don't think so but I remember reading this bit about the helpers?

https://github.com/DavidAnson/markdownlint/tree/b2305efafb034b1f328845aec9928b5363ffd646/helpers#api

> Undocumented - This package exports the internal functions as-is. The APIs were not originally meant to be public, are not officially supported, and may change from release to release.

Not really sure the best way to handle things or if I'm maybe misunderstanding something 😅
